#!/usr/bin/env python
PACKAGE = "phobos_control_manipulator"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("linear_increment",    double_t,    0, "Keyboard linear increment", 0.01,  0, 1)
gen.add("angular_increment", double_t, 0, "Keyboard angular increment",    0.01, 0,   1)
gen.add("joystick_sensitivity", double_t, 0, "Joystick sensitivity",    0.05, 0,   0.3)
gen.add("active", bool_t, 0, "If active", False);

size_enum = gen.enum([ gen.const("Joystick",      int_t, 0, "Joystick manipulator operation"),
                       gen.const("Keyboard",     int_t, 1, "Keyboard manipulator operation"),
                       ],
                     "Operation method")

gen.add("method", int_t, 0, "Parameter", 1, 0, 3, edit_method=size_enum)

exit(gen.generate(PACKAGE, "phobos_control_manipulator", "ManipParameters"))
